Rework build system

schlomo opened the pull request:
- build in local directory instead of user homedir
- use count of git revisions as version -> constantly increasing
- build via src.rpm so that production use can build rpm from src.rpm in mock environment
- add support for direct uploading to yum repo server (https://github.com/immobilienscout24/yum-repo-server)

schlomo committed:
Rework build system:
* build in local directory instead of user homedir
* use count of git revisions as version -> constantly increasing
* build via src.rpm so that production use can build rpm from src.rpm in mock environment
* add support for direct uploading to yum repo server (https://github.com/immobilienscout24/yum-repo-server)

albac commented:
@paulegan, I was recently looking at this and I have to say I am agree with @schlomo, or at least we should have options to build this base on any release/version or latest code and a good README instructions.

gdisantis24 committed:
[skip ci] Add Advanced Security code scanning workflow

advanced-security-actions-deployer[bot] committed:
[skip ci] Add Advanced Security code scanning workflow

[skip ci] Add Advanced Security code scanning workflow.
This file will introduce a new pipeline for the repository on GitHub, which doesn’t affect in any way already existing CI-CD pipelines.
If you have any questions please ask on #security channel.

schlomo commented:
Outdated
